pkg/spire/spire_gke.go

```diff
@@ -0,0 +1,58 @@
+package spire
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig"
+	"github.com/spiffe/go-spiffe/v2/workloadapi"
+	"io/fs"
+	"os"
+)
+
+const SPIFFE_ENDPT = "SPIFFE_ENDPOINT_SOCKET"
+
+func HasSpireEndpoint() bool {
+	s := os.Getenv(SPIFFE_ENDPT)
+	if s == "" {
+		return false
+	}
+	fi, err := os.Stat(s)
+	if err != nil {
+		return false
+	}
+	if fi.Mode().Type()&fs.ModeSocket != 0 {
+		return true
+	}
+	return false
+}
+
+func GetTlsClientConfig() (*tls.Config, error) {
+	s := os.Getenv(SPIFFE_ENDPT)
+	if s == "" {
+		return nil, fmt.Errorf(SPIFFE_ENDPT + " not found in environment")
+	}
+	src, err := workloadapi.NewX509Source(context.Background(), workloadapi.WithClientOptions(workloadapi.WithAddr("unix://"+s)))
```

Comment:
```suggestion
	src, err := workloadapi.NewX509Source(context.Background())
```
two things here, first the environ MUST be a URI, so the `"unix://"+s` would bread it.
> The value of the SPIFFE_ENDPOINT_SOCKET environment variable is structured as an RFC 3986 URI. The scheme MUST be set to either unix or tcp, which indicates that the endpoint is served over a Unix Domain Socket or a TCP listen socket, respectively.
https://github.com/spiffe/spiffe/blob/main/standards/SPIFFE_Workload_Endpoint.md#4-locating-the-endpoint

second you don't have to pass the withAddr option as that is done implicitly. but nothing worng with being explicit. 

>WithAddr provides an address for the Workload API. The value of the SPIFFE_ENDPOINT_SOCKET environment variable will be used if the option is unused. 
https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/workloadapi#WithAddr

also in both call sites to this function the getenv is checked before calling GetTLSFooConfig so you can remove them duplicate checks.